### PR TITLE
Add DebugKit documentation

### PR DESCRIPTION
**Pull request by markstory**
Having all the core plugin documentation in the book was something @lorenzo and I discussed briefly. The goals of doing this would be to centralize the documentation and hopefully make core plugins more discoverable.

If this is acceptable, I'll move the docs for migrations over as well.

### CODE REVIEW

**Comment by dakota** [comment]
:+1: good idea

**Comment by antograssiot** [comment]
That's a good idea :+1: and push some modification as a commit.

I think that a future improvement might be to add small gif "video" to help to explain the history panel works, with the color change and so one like it was done [here]( https://github.com/cakephp/debug_kit/pull/239) because I think that many people don't know how to access a SQL log for example if there's a redirect in their function.
I can understand that this kind of images can be quickly outdated but I think that it might provides a more user-friendly and attractive documentation. Ideas ?

**Comment by markstory** [comment]
I could include a few gifs of uaing the history panel as that seems to confuse people the most.

**Comment by lorenzo** [comment]
Having those gifs would be awesome!

File: en/contents.rst [hunk 1]
```diff
@@ -57,6 +57,7 @@ Contents
     core-libraries/xml
 
     core-libraries/global-constants-and-functions
+    debug-kit
```
Comment (lorenzo) [inline en/contents.rst]: I would move this after "Debugging"

**Comment by markstory** [comment]
Images and history panel data added.

File: en/contents.rst (hunk 1 above)
Comment (markstory) [inline en/contents.rst]: I thought about that too, but I feel like as we add more of the plugins having them grouped together outside of the core docs might be nice.

**Comment by lorenzo** [comment]
Great job!

**Pull request merged**